devito/mpi/halo_scheme.py

```diff
@@ -95,8 +127,16 @@ def __init__(self, exprs, ispace):
         self._honored = frozendict(self._honored)
 
     def __repr__(self):
```

Comment:
Could this be attached to a mixin class for both `HaloSpot` and `HaloSchemeEntry`?

Reply:
yes but where would you add it? 

I would just drop all these __repr__ it's a lot of code for what at the end of the day?

Reply:
When you print the IET, you see more useful details, it makes dev/debug life easier!

Reply:
anyway, Ed has a point, this is redundant code which should somehow be factorized (a private method of HaloScheme, also called by `HaloSpot.__repr__`? not sure)

Reply:
What I am trying to say is that in order to factor code out, someone has to factor out the fmapper/functions, but I had failed to do it. Now I think I did it, fix coming